### PR TITLE
[Do not merge; Testing] Reintroduce reshaped dist to check perf regression

### PR DESCRIPTION
**Pull request by neerajprad**
Refer to the discussion in #1544. This may not be needed for all distributions but I want to get our CI runtimes with this change. cc. @fehiepsi

### CODE REVIEW

File: pyro/distributions/torch_distribution.py [hunk 1]
```diff
@@ -221,6 +254,142 @@ def check(self, value):
...
+
+    def sample(self, sample_shape=torch.Size()):
+        return self.base_dist.sample(sample_shape + self.sample_shape)
+
+    def rsample(self, sample_shape=torch.Size()):
+        return self.base_dist.rsample(sample_shape + self.sample_shape)
+
+    def log_prob(self, value):
+        shape = broadcast_shape(self.batch_shape, value.shape[:value.dim() - self.event_dim])
+        return sum_rightmost(self.base_dist.log_prob(value), self.reinterpreted_batch_ndims).expand(shape)
```
Comment (fehiepsi) [inline pyro/distributions/torch_distribution.py]: This should be the way to do! Expand parameters first, then computing log_prob might be ineffective if log_prob contains expensive operators.

cc @fritzo for a throughout discussion about this problem

**Comment by fehiepsi** [comment]
I know the bug. Will make a PR at pytorch soon.

**Comment by neerajprad** [comment]
> I know the bug. Will make a PR at pytorch soon.

The PR will likely not make it into release. So could you patch the mvn distribution in `torch.py` in Pyro as well?

**Comment by neerajprad** [comment]
> Sounds great, thanks @fehiepsi.

File: pyro/distributions/torch_distribution.py (hunk 1 above)
Comment (neerajprad) [reply pyro/distributions/torch_distribution.py]: At a primary glance, it doesn't seem like we should be introducing the ReshapedDistribution back, but instead patch the mvn and lowrank mvn distributions, as you have done below. I'll gather a few test statistics and close this PR.

**Comment by neerajprad** [comment]
Closing this, as #1588 resolves the slowdown with a change localized to mvn rather than reintroducing `ReshapedDistribution`.

**Pull request closed without merging**